common/src/com/thoughtworks/go/domain/materials/Modifications.java

```diff
@@ -129,6 +129,11 @@ public boolean shouldBeIgnoredByFilterIn(MaterialConfig materialConfig) {
             return false;
         }
         Set<ModifiedFile> allFiles = getAllFiles(this);
+
+        if (allFiles.isEmpty()) {
```

Comment:
Thinking about this a bit more, won't this allow a normal git merge commit to actually trigger a build, when the commit should be ignored because of file modifications? What I mean is: modificationSince returns two commits. First is a commit whose filter should cause the commit to be ignored. Second is a merge commit. Earlier, this would be an ignored set of commits and a build wouldn't run for this. Now, it will.